client/macro_processor.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import fnmatch
+import os
+import re
+import shutil
+import yaml
+
+
+from argparse import Namespace
+from yaml.loader import SafeLoader
+from os.path import dirname, isfile, join
+from typing import Dict
+
+
+class MacrosProcessor:
+    """A processor to handle macros in the query files during the pre-processing and post-processing stages of a Batch
+    Sql Translation job.
+
+    """
+    def __init__(self, context: Namespace):
+        self.context = context
+        if context.macro_map:
+            self.enable_macro_substitution = True
+            self.macros_subst = MapBasedExpander(context.macro_map)
+        else:
+            self.enable_macro_substitution = False
+
+    def pre_process(self, input_dir: str, output_dir: str):
```

Comment:
I think no _ in preprocess, it's not common.

Reply:
Done.